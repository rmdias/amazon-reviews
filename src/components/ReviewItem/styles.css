.arw_reviews__item {
  padding: 20px;
  margin-bottom: 15px;
}

.arw_reviews__item_header {
  display: flex;
}

.arw_reviews__item_info {
  display: flex;
}

.arw_reviews__item_info_column {
  padding: 0 7%;
}

.arw_reviews__item_info_column:last-child {
  padding-right: 0;
}

.arw_reviews__item_name {
  max-width: 140px;
  display: inline-block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

span.arw_reviews__item_info__title {
  font-size: 11px;
  display: block;
  padding-bottom: 4px;
  text-transform: uppercase;
}

.arw_reviews__title {
  font-size: 16px;
  padding: 25px 0 5px 0;
}

@media only screen and (max-width: 768px) {
  .arw_reviews__item_header {
    display: block;
  }

  img.arw_reviews__item_image {
    width: 100%;
    height: 150px;
  }

  div.arw_reviews__item_info {
    display: flex;
    margin-top: 20px;
  }

  div.arw_reviews__item_info_column {
    padding: 0 2%;
  }

  .arw_reviews__item_info_column:first-child {
    padding-left: 0;
  }
}