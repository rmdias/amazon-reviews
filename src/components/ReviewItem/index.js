import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Area, Hint, Image as ProductImage } from 'luna-ui-lib'

import './styles.css'

export default class Reviews extends PureComponent {
  static propTypes = {
    review: PropTypes.object
  }

  render() {
    return (
      <Area className="arw_reviews__item">
        <header className="arw_reviews__item_header">
          <ProductImage
            fit
            shadow
            width="80px"
            height="80px"
            className="arw_reviews__item_image"
            src={`https://images-na.ssl-images-amazon.com/images/I/${this.props.review.productImg}._SY679_.jpg`}
          />

          <div className="arw_reviews__item_info">
            <Hint className="arw_reviews__item_info__title">date</Hint>
            <strong>{this.props.review.created}</strong>
          </div>

          <div className="arw_reviews__item_info">
            <Hint className="arw_reviews__item_info__title">stars</Hint>
            <strong>{this.props.review.stars} ★★★★</strong>
          </div>

          <div className="arw_reviews__item_info">
            <Hint className="arw_reviews__item_info__title">{this.props.review.reviewId}</Hint>
            <strong className="arw_reviews__item_name">{this.props.review.productTitle}</strong>
          </div>
        </header>
        <h4 className="arw_reviews__title"><strong>{this.props.review.title}</strong></h4>
        <p>{this.props.review.content}</p>
      </Area>
    )
  }
}
