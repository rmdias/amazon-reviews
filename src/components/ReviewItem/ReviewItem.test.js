import React from 'react'
import renderer from 'react-test-renderer'

import ReviewItem from './index'
import ReviewMock from './__mocks__/__mock__review'

import { Theme } from 'luna-ui-lib'

describe('<ReviewItem />', () => {
  it('toMatchSnapshot: Load <ReviewItem /> component', () => {
    const component = <Theme><ReviewItem review={ReviewMock}/></Theme>

    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
