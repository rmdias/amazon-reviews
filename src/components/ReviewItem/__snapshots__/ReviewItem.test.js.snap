// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewItem /> toMatchSnapshot: Load <ReviewItem /> component 1`] = `
<div>
  <div
    className="arw_reviews__item sc-kpOJdX ddRiWz"
  >
    <header
      className="arw_reviews__item_header"
    >
      <img
        alt="image"
        className="arw_reviews__item_image sc-dxgOiQ kgxBsP"
        src="https://images-na.ssl-images-amazon.com/images/I/undefined._SY679_.jpg"
      />
      <div
        className="arw_reviews__item_info"
      >
        <div
          className="arw_reviews__item_info_column"
        >
          <span
            className="arw_reviews__item_info__title sc-ckVGcZ GYKlq"
          >
            date
          </span>
          <strong>
            07.02.2018
          </strong>
        </div>
        <div
          className="arw_reviews__item_info_column"
        >
          <span
            className="arw_reviews__item_info__title sc-jKJlTe cOhKeT"
          >
            stars
          </span>
          <strong>
            ★★★★★
          </strong>
        </div>
        <div
          className="arw_reviews__item_info_column"
        >
          <span
            className="arw_reviews__item_info__title sc-eNQAEJ dWsydp"
          />
          <strong
            className="arw_reviews__item_name"
          />
        </div>
      </div>
    </header>
    <h4
      className="arw_reviews__title"
    >
      <strong>
        Nice rattle sound
      </strong>
    </h4>
    <p />
  </div>
</div>
`;
